common/remote/sockfile.cpp

```diff
@@ -5597,26 +5805,46 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
         {
             responseWriter.setown(createIXmlWriterExt(0, 0, nullptr, outFmt_Xml == outputFormat ? WTStandard : WTJSON));
             responseWriter->outputBeginNested("Response", true);
-            responseWriter->outputUInt(cursorHandle, sizeof(cursorHandle), "cursor");
+            responseWriter->outputUInt(cursorHandle, sizeof(cursorHandle), "handle");
         }
         if (cursorHandle)
         {
             IOutputMetaData *out = outputActivity->queryOutputMeta();
-            unsigned __int64 initProcessed = outputActivity->queryProcessed();
+            bool grouped = outputActivity->isGrouped();
             bool eoi=false;
             if (outFmt_Binary == outputFormat)
             {
                 DelayedSizeMarker dataLenMarker(reply); // data length
+                outputActivity->setBuilderBuffer(reply);
                 for (unsigned __int64 i=0; i<defaultDaFSNumRecs; i++)
                 {
+                    bool *eog;
+                    if (grouped && outputActivity->queryProcessed()) // i.e. not 1st row
+                    {
+                        eog = (bool *)reply.reserve(1);
+                        *eog = outputActivity->getClearEogPending(); // NB: can change if false, there are rows in group and last row of group is filtered
+                    }
                     size32_t rowSz;
-                    const void *row = outputActivity->nextRow(rowSz);
+                    const void *row = outputActivity->nextRow(rowSz); // NB: row builder writes directly to reply buffer for effiency
                     if (!row)
                     {
-                        eoi = true;
-                        break;
+                        if (grouped)
+                        {
+                            if (outputActivity->queryProcessed()) // i.e. not 1st row
```

Comment:
This seems a strange test to see if it is the first row - I would expect testing a local variable e.g., i != 0 or